scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -977,17 +978,37 @@ void pingTaskNode(RunningTaskData taskData) {
         } catch (Throwable t) {
             RunningTaskData runningTask = jobs.getRunningTask(task.getId());
             if (runningTask != null) {
-                int attempts = runningTask.increaseAndGetPingAttempts();
-                if (attempts > PASchedulerProperties.SCHEDULER_NODE_PING_ATTEMPTS.getValueAsInt()) {
-                    tlogger.info(task.getId(), "node failed", t);
-                    restartTaskOnNodeFailure(task);
+                // We try to ping the node where the task is running.
+                // We don't consider here other nodes reserved for the task, 
+                // as it is the responsibility of the task itself to manage extra nodes lifecycle 
+                // in case of complex multinodes task deployment.
+                if (!pingNode(runningTask.getNodes().get(0))) {
```

Comment:
The first node in the list is the node used to execute the task. This is how the NodeSet is built. Maybe there's a way to make it more explicit, I will have a look but it may be a bit longer to implement.